pom.xml

```diff
@@ -1848,5 +1858,142 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>quick</id>
+            <activation>
+                <property>
+                    <name>quick</name>
+                </property>
+            </activation>
+            <properties>
+                <skipTests>true</skipTests>
+                <maven.test.skip>true</maven.test.skip>
```

Comment:
test compilation skipping is not useful in practice IMO. We change as much test code as product code in each PR.

In my local bash aliases I skip the source and javadoc steps though.

Reply:
I think we should use this profile on the CI to compile before tests execution. That way it will be useful and possibly we could simplify `MAVEN_FAST_INSTALL` env variable.

Also we could possibly extract a shell script that could be used do a quick build with all the magic (maybe with some caching in future).

Reply:
caching on CI is already here since https://github.com/trinodb/trino/pull/10115.

Agreed about MAVEN_FAST_INSTALL. 👍

Reply:
The goal here is to be able to run the server as quickly as possible. Compiling tests is not required for this. It's useful in some CI jobs that separate running checks and running tests, to detect compilation failures earlier, but I think we should skip them here.

Reply:
Got it. So `maven-checks` does the full checks so that other jobs can skip them and finish faster.

Reply:
I was considering disabling some modules in this profile, but it would require splitting up the list of modules so that some are added in a default profile _unless_ quick is enabled. Would that be ok? Or should we discuss this in a follow-up PR? The benchmark and verifier modules take 7 seconds to build. I excluded docs and rpm from all my measurements anyway.

@kokosing the point of having this profile is that you don't need any other Maven flags, other than module selectors, so maybe a script is not necessary?

Reply:
> so maybe a script is not necessary?

i hope we don't end up with a script. 

Script works great for a silo'ed command, one that you never need to modify, which is not what engineers need on the CLI. I sometimes need to build one module, sometimes two, sometimes the whole project. 

> I was considering disabling some modules in this profile, but it would require splitting up the list of modules so that some are added in a default profile unless quick is enabled. Would that be ok? Or should we discuss this in a follow-up PR? The benchmark and verifier modules take 7 seconds to build. I excluded docs and rpm from all my measurements anyway.

Let's follow up.

Reply:
Question: would that be still possible to do `-Pquick -Dmaven.test.skip=false` to have all the benefits of quick compilation - but still compile test code too?

Reply:
I had to split this into two separate profiles:
* quick - disables checks, javadoc, site and sources
* noTests - doesn't compile tests and disable the dependency plugin

Without disabling the dependency plugin, build fails when `~/.m2/repository` is empty since dependency analysis is executed even when tests are not being built and requires artifacts from all scopes to be resolvable.

I'll resolve this thread since the change is outdated. Please comment on the new profiles if you think they're useful or not. We don't need to keep them in the project, devs could define them in `%USER_HOME%/.m2/settings.xml`

Reply:
Also, I'll update the PR description with new benchmarks.

Reply:
> I had to split this into two separate profiles:
> 
> quick - disables checks, javadoc, site and sources
> noTests - doesn't compile tests and disable the dependency plugin

Sounds like a complication. Do we really need it?

In the worst case, there will be occasions where `-Pquick` cannot be used, right?

Does this also mean that `-Pquick -PnoTests` is faster than `-Pquick` alone?

Reply:
Aaand turns out that the shade plugin also triggers full dependency resolution, so there's no way of actually fully disabling test compilation.